src/chains/ethereum/ethereum/src/miner/executables.ts

```diff
@@ -1,7 +1,42 @@
 import { TypedTransaction } from "@ganache/ethereum-transaction";
 import { Heap } from "@ganache/utils";
+import { byNonce } from "../transaction-pool";
 
-export type Executables = {
-  inProgress: Set<TypedTransaction>;
-  pending: Map<string, Heap<TypedTransaction>>;
-};
+export class Executables {
+  public inProgress: Set<TypedTransaction> = new Set();
+  public pending: Map<string, Heap<TypedTransaction>> = new Map();
+
+  /**
+   * Deep clones the executables, moving all `inProgress` transactions to the
```

Comment:
I initially didn't think that this clones the transactions themselves, but after looking at the code I see that it does.

Maybe worth making this more clear in the comment - e.g., `... cloning all transactions into "pending" queue ..."

Reply:
Clarified!